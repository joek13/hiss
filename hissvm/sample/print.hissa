# validates print format

.constants {
    hfunc 0 $main
    hfunc 2 $add
    hint 42
}

.code {
main:
    pushc 0
    print
    pushc 1
    print
    pushc 2
    print
    halt
add:
    loadv 0x00
    loadv 0x01
    iadd
    ret
}