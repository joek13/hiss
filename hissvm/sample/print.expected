<0-ary fn @ 0x00>
<2-ary fn @ 0x0A>
42